r2-netty/src/main/java/com/linkedin/r2/netty/client/http/HttpChannelPoolFactory.java

```diff
@@ -90,8 +96,19 @@ public HttpChannelPoolFactory(
     _tcpNoDelay = tcpNoDelay;
     _channelPoolWaiterTimeout = channelPoolWaiterTimeout;
 
-    _bootstrap = new Bootstrap().group(eventLoopGroup).channel(NioSocketChannel.class).
-        option(ChannelOption.CONNECT_TIMEOUT_MILLIS, connectTimeout).handler(initializer);
+    if (!StringUtils.isEmpty(_udsAddress)) {
+      _bootstrap = new Bootstrap().
+          group(new EpollEventLoopGroup()).
```

Comment:
Could this event loop group be injected from the factory? Should it be different from `eventLoopGroup`?
In `ChannelPoolManagerFactoryImpl` it says the creator of the factory has responsibility for shutting down the event loop group; I don't see this one being shut down?

Reply:
This is a great point, that you called out about shutdown. Thank you. I have tried to address it two ways:
- Changes to the HttpClientFactory class, where the default would be `EpollEventLoopGroup`
- Added a change in `container` lib to pass in `EpolllEventLoopGroup` as well. This should hopefully address the problem. Please do let me know if I can handle this in a better way.